### DISCUSSION TITLE
How do I add a system (Virtual Boy)?

### DISCUSSION THREAD

**Discussion by delacroix17** [rg35xx-cfw/Koriki#44] [Q&A] [answered]
I checked in handhelds and consoles and Virtual Boy is missing, even though I have games there. I am able to find it if I go through Retroarch, but is there a way to make it appear in the main menus? 

The .ini file for Consoles says:
[VIRTUAL BOY]
execs = /userdata/system/.simplemenu/launchers/mednafen_vb_libretro
romDirs = /userdata/roms/virtualboy,/userdata/Roms/VB/
romExts = .vb,.vboy,.bin,.zip,.7z

Could "mednafen_vb_libretro" be the wrong name? Because I don't see it in the "launchers" folder.  

I tried messing with these a bit, and I'm not even sure that this file is what controls what is in the home page.

**Comment by delacroix17** [comment]
Okay everyone, I think I figured it out. It simply needed a "/" after "/roms/virtualboy" in the console.ini file.

**Reply by delacroix17** [reply]
It seems I've broken it again. While I can access the Virtual Boy selection menu now, games won't boot up. If I leave "vb_libretro" in the launchers folder as is, I get an error saying "cannot find mednafen_vb_retro." But if I change the launcher name to "mednafen_vb_libretro" then I simply get a black screen and goes back to game selection menu. 

Booting through retroarch still works normally.

**Reply by delacroix17** [reply, marked as answer]
My ugly fix to everything is to create 2 launchers, "vb_libretro" and "mednafen_vb_libretro" and while selecting a game make sure to choose "vb_libretro." 
I'm sure there's some step inbetween that interfaces koriki with retroarch, but I don't know how to edit that, so this will have to do until someone teaches me.

**Reply by acmeplus** [reply]
you can also change which is the default launcher but putting the launcher entry as the first for a system. For instance, instead of:
```[VIRTUAL BOY]
execs = /userdata/system/.simplemenu/launchers/vb_libretro
romDirs = /userdata/roms/virtualboy/,/userdata/Roms/VB/
romExts = .vb,.vboy,.bin,.zip,.7z
```

you could have: 
```[VIRTUAL BOY]
execs = /userdata/system/.simplemenu/launches/mednafen_vb_libretro,/userdata/system/.simplemenu/launchers/vb_libretro
romDirs = /userdata/roms/virtualboy/,/userdata/Roms/VB/
romExts = .vb,.vboy,.bin,.zip,.7z
``` 
so ``mednafen_vb_libretro`` would be the default launcher.
